client/src/components/ColorList.js

```diff
@@ -1,32 +1,65 @@
-import React, { useState } from "react";
-import axios from "axios";
+import React, { useState, useEffect } from "react";
+import { axiosWithAuth } from "../utils/axiosWithAuth";
+import axios from 'axios';
 
 const initialColor = {
   color: "",
   code: { hex: "" }
 };
 
-const ColorList = ({ colors, updateColors }) => {
-  console.log(colors);
+const ColorList = ({ colors, updateColors, props, setColorList }) => {
+  console.log('from colorList', props);
   const [editing, setEditing] = useState(false);
   const [colorToEdit, setColorToEdit] = useState(initialColor);
 
+  // console.log(colorToEdit)
+
   const editColor = color => {
     setEditing(true);
     setColorToEdit(color);
   };
 
+  useEffect(() => {
+
+      const id = props.match.params.id;
+      console.log('ID', id)
+
+      const colorsList = colors.find( color => `${color.id}` === id);
+
+    if(colorsList) setColorToEdit(colorsList)
+  }, [colors, props.match.params.id])
+
   const saveEdit = e => {
     e.preventDefault();
     // Make a put request to save your updated color
     // think about where will you get the id from...
     // where is is saved right now?
+    axiosWithAuth()
+      .put(`http://localhost:5000/api/colors/${colorToEdit.id}`, colorToEdit) 
```

Comment:
this should still work with the colorToEdit removed since you are inserting the colorToEdit.id